tests/src/item-namespace-staking.test.ts

```diff
@@ -0,0 +1,748 @@
+      itemMintKeypair.publicKey,
+      new anchor.BN(itemIndex)
+    );
+
+    const accInfo = await itemProgram.client.account.item.fetch(
+      itemPDA,
+      "confirmed"
+    );
+
+    expect(accInfo).toBeDefined();
+
+    expect((accInfo.classIndex as anchor.BN).toNumber()).toBe(itemClassIndex);
+  });
+
+  it("Mints namespace NFT", async () => {
+    await mintNFT(provider, namespaceMintKeypair);
+
+    const tokenAccounts =
+      await provider.connection.getParsedTokenAccountsByOwner(
+        walletKeypair.publicKey,
+        {
+          mint: namespaceMintKeypair.publicKey,
+        },
+        "confirmed"
+      );
+
+    expect(tokenAccounts.value.length).toBe(1);
+  });
+
+  it("Mints staking tokens", async () => {
+    await mintTokens(provider, stakingMintKeypair, stakingAmount, 0);
+
+    const tokenAccounts =
+      await provider.connection.getParsedTokenAccountsByOwner(
+        walletKeypair.publicKey,
+        {
+          mint: stakingMintKeypair.publicKey,
+        },
+        "confirmed"
+      );
+
+    expect(tokenAccounts.value.length).toBe(1);
+
+    expect(
+      tokenAccounts.value[0].account.data.parsed.info.tokenAmount.uiAmount
+    ).toBe(stakingAmount);
+  });
+
+  it("Initializes a namespace", async () => {
+    const namespaceMintMetadataPDA = await Utils.PDA.getMetadata(
+      namespaceMintKeypair.publicKey
+    );
+
+    const namespaceMintMasterEditionPDA = await Utils.PDA.getEdition(
+      namespaceMintKeypair.publicKey
+    );
+
+    const args = {
+      desiredNamespaceArraySize: new anchor.BN(2),
+      uuid: "e3e52d",
+      prettyName: "Trash with Frens",
+      permissivenessSettings: {
+        namespacePermissiveness: State.Namespace.Permissiveness.All,
+        itemPermissiveness: State.Namespace.Permissiveness.All,
+        playerPermissiveness: State.Namespace.Permissiveness.All,
+        matchPermissiveness: State.Namespace.Permissiveness.All,
+        missionPermissiveness: State.Namespace.Permissiveness.Namespace,
+        cachePermissiveness: State.Namespace.Permissiveness.All,
+      },
+      whitelistedStakingMints: [stakingMintKeypair.publicKey],
+    } as Instructions.Namespace.InitializeNamespaceArgs;
+
+    const accounts = {
+      mint: namespaceMintKeypair.publicKey,
+      metadata: namespaceMintMetadataPDA,
+      masterEdition: namespaceMintMasterEditionPDA,
+    } as Instructions.Namespace.InitializeNamespaceAccounts;
+
+    await namespaceProgram.initializeNamespace(args, accounts);
+
+    const [namespacePDA] = await Utils.PDA.getNamespacePDA(
+      namespaceMintKeypair.publicKey
+    );
+
+    const accInfo = await namespaceProgram.client.account.namespace.fetch(
+      namespacePDA,
+      "confirmed"
+    );
+
+    expect(accInfo).toBeDefined();
+
+    expect((accInfo.mint as anchor.web3.PublicKey).toBase58()).toBe(
+      namespaceMintKeypair.publicKey.toBase58()
+    );
+  });
+
+  it("Creates namespace gatekeeper", async () => {
```

Comment:
I already have namespace tests like this, I dont think you need to rewrite them

Reply:
https://github.com/raindrops-protocol/raindrops/blob/master/rust/tests/namespace.ts

Reply:
It's not an individual test. For item staking, item must be joined to a namespace and to join namespace, it must have namespace gatekeeper. So we should create namespace gatekeeper first here.